packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -1023,10 +1023,24 @@ function* recoverPurchaseIfNecessary() {
       route: quote.route
     })
 
-    // Check if we have an exchangable amount of SOL, and if so, exchange it to AUDIO
+    // Subtract fees and rent to see how much SOL is available to exchange
     const exchangableBalance = new BN(existingBalance).sub(totalFees)
-    // Usually indicates Swap Failed
-    if (exchangableBalance.gt(new BN(0))) {
+
+    // Use proportion to guesstimate an $AUDIO quote for the exchangeable balance
+    const estimatedAudio =
+      existingBalance > 0
+        ? exchangableBalance
+            .mul(new BN(quote.outputAmount.amountString))
+            .div(new BN(existingBalance))
+        : new BN(0)
+
+    // Check if there's a non-zero exchangeble amount of SOL and at least one $AUDIO would be output
+    // Should only occur as the result of a previously failed Swap
+    if (
+      exchangableBalance.gt(new BN(0)) &&
```

Comment:
since `estimatedAudio` is already based on` exchangableBalance`, do we need to explicitly check for `exchangeableBalance > 0` here?

Reply:
Trying to avoid a divide by zero

Reply:
ah, no